apps/sms/views/new_message/js/new_message.js

```diff
@@ -0,0 +1,3031 @@
+function conv_generateSmilSlides(slides, content) {
+  var length = slides.length;
+  if (typeof content === 'string') {
+    if (!length || slides[length - 1].text) {
+      slides.push({
+        text: content
+      });
+    } else {
+      slides[length - 1].text = content;
+    }
+  } else {
+    slides.push({
+      blob: content.blob,
+      name: content.name
+    });
+  }
+  return slides;
+}
+
+var ConversationView = {
+  CHUNK_SIZE: 10,
+  // duration of the notification that message type was converted
+  CONVERTED_MESSAGE_DURATION: 3000,
+  IMAGE_RESIZE_DURATION: 3000,
+  BANNER_DURATION: 2000,
+
+  // Toast duration when you write a long text and need more than one SMS
+  // to send it
+  ANOTHER_SMS_TOAST_DURATION: 3000,
+
+  // when sending an sms to several recipients in activity, we'll exit the
+  // activity after this delay after moving to the thread list.
+  LEAVE_ACTIVITY_DELAY: 3000,
+
+  draft: null,
+  recipients: null,
+  // Set to |true| when in edit mode
+  inEditMode: false,
+  isNewMessageNoticeShown: false,
+  shouldChangePanelNextEvent: false,
+  showErrorInFailedEvent: '',
+  previousSegment: 0,
+
+  timeouts: {
+    update: null,
+    subjectLengthNotice: null
+  },
+
+  multiSimActionButton: null,
+  init: function conv_init() {
+    var templateIds = [
+      'message',
+      'message-sim-information',
+      'message-status',
+      'not-downloaded',
+      'recipient',
+      'date-group',
+      'header',
+      'group-header'
+    ];
+
+    // Fields with 'messages' label
+    [
+      'container', 'to-field', 'recipients-list', 'compose-form', 'header',
+      'edit-header', 'check-uncheck-all-button', 'contact-pick-button',
+      'send-button', 'delete-button', 'call-number-button', 'options-button',
+      'new-message-notice', 'edit-mode', 'edit-form', 'header-text',
+      'max-length-notice', 'convert-notice', 'resize-notice',
+      'new-message-notice', 'subject-max-length-notice', 'sms-counter-notice',
+      'recipient-suggestions'
+    ].forEach(function(id) {
+      this[Utils.camelCase(id)] = document.getElementById('messages-' + id);
+    }, this);
+
+    this.mainWrapper = document.getElementById('main-wrapper');
+    this.threadMessages = document.getElementById('thread-messages');
+
+    window.addEventListener('resize', this.resizeHandler.bind(this));
+
+    // binding so that we can remove this listener later
+    this.onVisibilityChange = this.onVisibilityChange.bind(this);
+    document.addEventListener('visibilitychange',
+                              this.onVisibilityChange);
+
+    this.toField.addEventListener(
+      'keypress', this.toFieldKeypress.bind(this), true
+    );
+
+    this.toField.addEventListener(
+      'input', this.toFieldInput.bind(this), true
+    );
+
+    this.toField.addEventListener(
+      'focus', this.toFieldInput.bind(this), true
+    );
+
+    this.sendButton.addEventListener(
```

Comment:
Looks like we don't need any handlers on sendButton anymore, since we ```activate``` Compose event that does the same (assimilate recipients). And in case something goes wrong and button is clicked when it should have been disabled, MessagingService should prevent sending.